android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java

```diff
@@ -82,7 +87,14 @@ protected void onCreate(Bundle savedInstanceState) {
             showUserData(user);
         });
 
-        avatar.setOnClickListener(v -> showFileChooser());
+
+        avatar.setOnClickListener(v -> {
+            if(checkReadExternalPermission()) {
+                showFileChooser();
```

Comment:
Here there is a check where the app checks whether external storage permission is granted or not. So when the user clicks the profile pic and denies permission file chooser is not fired.

Reply:
So to update the pic the user needs to grant the app external permission. Do correct me if I am not able to get your point at all !